apis/v1alpha1/adminnetworkpolicy_types.go

```diff
@@ -0,0 +1,174 @@
+}
+
+// AdminNetworkPolicyStatus defines the observed state of AdminNetworkPolicy.
+type AdminNetworkPolicyStatus struct {
+	Conditions []metav1.Condition `json:"conditions"`
+}
+
+// AdminNetworkPolicySpec defines the desired state of AdminNetworkPolicy.
+type AdminNetworkPolicySpec struct {
+	// Priority is an int32 value bound to 0 - 1000, the lowest priority,
+	// "0" corresponds to the highest importance, while higher priorities have
+	// lower importance.
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=1000
+	Priority int32 `json:"priority"`
+
+	// Subject defines the pods to which this AdminNetworkPolicy applies.
+	Subject AdminNetworkPolicySubject `json:"subject"`
+
+	// List of Ingress rules to be applied to the selected pods BEFORE any
+	// NetworkPolicy or BaselineAdminNetworkPolicy rules have been applied.
+	// A total of 100 rules will be allowed in each ANP instance.
+	// ANPs with no ingress rules do not affect ingress traffic.
+	// +optional
+	// +kubebuilder:validation:MaxItems=100
+	Ingress []AdminNetworkPolicyIngressRule `json:"ingress,omitempty"`
+
+	// List of Egress rules to be applied to the selected pods BEFORE any
+	// NetworkPolicy or BaselineAdminNetworkPolicy rules have been applied.
+	// A total of 100 rules will be allowed in each ANP instance.
+	// ANPs with no egress rules do not affect egress traffic.
+	// +optional
+	// +kubebuilder:validation:MaxItems=100
+	Egress []AdminNetworkPolicyEgressRule `json:"egress,omitempty"`
+}
+
+// AdminNetworkPolicyIngressRule describes an action to take on a particular
+// set of traffic destined for pods selected by an AdminNetworkPolicy's
+// Subject field. The traffic must match both ports and from.
+type AdminNetworkPolicyIngressRule struct {
+	// Name is an identifier for this rule, that may be no more than 100 characters
+	// in length. This field should be used by the implementation to help
+	// improve observability, readability and error-reporting for any applied
+	// AdminNetworkPolicies.
+	// +optional
+	// +kubebuilder:validation:MaxLength=100
+	Name string `json:"name,omitempty"`
+
+	// Action specifies the affect this rule will have on matching traffic,
+	// currently the following actions are supported:
+	// Allow: allows the selected traffic
+	// Deny: denies the selected traffic
+	// Pass: instructs the selected traffic to skip any remaining ANP rules, and
+	// then pass execution to any NetworkPolicies that select the pod.
+	// If the pod is not selected by any NetworkPolicies then execution
+	// is passed to any BaselineAdminNetworkPolicies that select the pod.
+	// This field is mandatory.
+	Action AdminNetworkPolicyRuleAction `json:"action"`
+
+	// Ports allows for matching traffic based on port and protocols.
+	// If Ports is not set then traffic is not filtered via port.
+	// +optional
+	// +kubebuilder:validation:MaxItems=100
+	Ports []AdminNetworkPolicyPort `json:"ports,omitempty"`
```

Comment:
Urg if we do that we end up with the 

```
ports: 
    ports: 
    -
```
So I'll use `*[]AdminNetworkPolicyPort` for now and investigate further if this will work

Reply:
Went with `*[]AdminNetworkPolicyPort` For now until I can do some research